homeassistant/components/dhcp/__init__.py

```diff
@@ -68,22 +68,17 @@ class DhcpServiceInfo(BaseServiceInfo):
     hostname: str
     macaddress: str
 
-    # Used to prevent log flooding. To be removed in 2022.6
-    _warning_logged: bool = False
-
     def __getitem__(self, name: str) -> Any:
         """
         Enable method for compatibility reason.
 
         Deprecated, and will be removed in version 2022.6.
         """
-        if not self._warning_logged:
-            report(
-                f"accessed discovery_info['{name}'] instead of discovery_info.{name}; this will fail in version 2022.6",
-                exclude_integrations={DOMAIN},
-                error_if_core=False,
-            )
-            self._warning_logged = True
+        report(
+            f"accessed discovery_info['{name}'] instead of discovery_info.{name}; this will fail in version 2022.6",
+            exclude_integrations={DOMAIN},
+            error_if_core=False,
+        )
         return getattr(self, name)
 
```

Comment:
In a followup PR we can probably collapse some of these into the base class, although `exclude_integrations` might be an issue.

Reply:
I considered it, and I could add all five domains (`mqtt`/`usb`/`ssdp`/`zeroconf`/`dhcp`) to `excluded_integrations`.

But:
- ssdp would still need to override it.
- ssdp, dhcp and zeroconf would still need to add the extra `get` method
- ssdp would still need to add the extra `__contains__` method

And I thought also that new discovery methods wouldn't need that code.

Reply:
👍 

Probably not worth the change on this since its going to be removed anyways in a few months.